tests/upgrades/test_performance_tuning.py

```diff
@@ -0,0 +1,221 @@
+    Performance Tuning utility
+
+    Test Steps:
+
+    1. Before Satellite upgrade, we will apply the medium tune size.
+    2. Once the size get apply we will wait for post upgrade test case to check
+    whether the size is same or not.
+    3. In post upgrade we will check whether satellite installer was able to change
+    the size or not
+
+    """
+    @staticmethod
+    def _custom_hiera_file_creation(mongodb_type, tune_size="default"):
+        """
+        This function we will use to create a custom-hiera.yaml file
+        as per users requirement"
+        :param int mongodb_type: Use to select  MMPV1 or WiredTiger data.
+        :param str tune_size: tune size would be medium, large, extra-large,
+        extra-extra-large.
+        """
+        with open('custom-hiera.yaml', 'w') as fd:
+            fd.write("\n".join(DEFAULT_CUSTOM_HIERA_DATA))
+            if mongodb_type:
+                fd.write("\n".join(MMPV1_MONGODB))
+            else:
+                fd.write("\n".join(WIREDTIGER_MONGODB))
+            if tune_size == "medium":
+                fd.write("\n".join(MEDIUM_TUNING_DATA))
+
+    @pre_upgrade
+    def test_pre_custom_hiera_file_validation(self):
+        """ In preupgrade scenario we will validate the default custom hiera file
+        parameter with expected custom_hiera file.
+        """
+        actual_custom_hiera = ssh.command("cat /etc/foreman-installer/custom-hiera.yaml")\
+            .stdout
+        actual_custom_hiera = set([data.strip() for data in actual_custom_hiera
+                                   if data != ''])
+        cmd = 'grep "mongodb::server::storage_engine: \'wiredTiger\'" ' \
+              '/etc/foreman-installer/custom-hiera.yaml'
+        mongodb_type = ssh.command(cmd).return_code
+        if mongodb_type:
+            expected_custom_hiera = set(DEFAULT_CUSTOM_HIERA_DATA + MMPV1_MONGODB)
+        else:
+            expected_custom_hiera = set(DEFAULT_CUSTOM_HIERA_DATA + WIREDTIGER_MONGODB)
+
+        assert expected_custom_hiera == actual_custom_hiera
+
+    @pre_upgrade
+    def test_pre_performance_tuning_apply(self):
+        """In preupgrade scenario we will apply the medium tuning size.
+
+        :id: preupgrade-83404326-20b7-11ea-a370-48f17f1fc2e1
+
+        :steps:
+            1. Run satellite-installer --disable-system-checks.
+
+        :expectedresults: Medium tuning parameter should be applied.
+
+         """
+        cmd = 'grep "mongodb::server::storage_engine: \'wiredTiger\'" ' \
+              '/etc/foreman-installer/custom-hiera.yaml'
+        mongodb_type = ssh.command(cmd).return_code
+        self._custom_hiera_file_creation(mongodb_type, "medium")
+
+        ssh.upload_file('custom-hiera.yaml', '/etc/foreman-installer')
+        command_output = ssh.command('satellite-installer --disable-system-checks',
+                                     connection_timeout=1000).stdout
+        command_output = [data.strip(" ") for data in command_output]
```

Comment:
You need not to convert the outpu string into list, as you can directly use 'in' operator to check in output string. So remove this line and say:
assert "Success!" in command_output   (here command_output is string)

But the question is what if there are more than one Success / Failure ?

Reply:
Fixed